bookmarks/vanilla/src/modules/Bookmark.ts

```diff
@@ -183,9 +187,10 @@ export class Bookmark {
             thing = setUrl(thing, RDF.type, BOOKMARK.Bookmark)
 
             const updatedBookmarkList = setThing(ds, thing);
+
             await saveSolidDatasetAt(indexUrl, updatedBookmarkList, { fetch: session.fetch });
 
-            return { url, ...payload }
+            return thing ? this.mapBookmark(thing) : undefined
```

Comment:
The method definition doesn't say it can return undefined. But I still think the correct behavior here is to throw an exception.